File: solutions/easy/java/HappyNumber.java
import java.util.*;

class HappyNumber {
    HashSet ss = new HashSet();
    public boolean isHappy(int n) {
        int num = 0;
        ss.add(n);
        while(n >0 ){
            num += (n%10) * (n%10);
            n /= 10;
        }
        if (num ==1 )
            return true;
        else if (ss.contains(num))
            return false;
        else {
            return isHappy(num);
        }
    }
}

File: solutions/easy/kotlin/HappyNumber.kt
internal class Solution {
    var ss: HashSet = HashSet()
    fun isHappy(n: Int): Boolean {
        var num = 0
        ss.add(n)
        while (n > 0) {
            num += n % 10 * (n % 10)
            n /= 10
        }
        return when{
            num == 1 -> true
            ss.contains(num) -> false
            else -> isHappy(num)
        }
    }
}